src/util/fields.py

```diff
@@ -652,7 +653,66 @@ def convert_field_to_html(cr, model, field, skip_inherit=()):
         convert_field_to_html(cr, inh.model, field, skip_inherit=skip_inherit)
 
 
-def convert_field_to_property(
+def _convert_field_to_company_dependent(
+    cr, model, field, type, target_model=None, default_value=None, default_value_ref=None, company_field="company_id"
+):
+    _validate_model(model)
+    if target_model:
+        _validate_model(target_model)
+
+    type2field = {"char", "float", "boolean", "integer", "text", "many2one", "date", "selection", "html"}
+    assert type in type2field
+
+    table = table_of_model(cr, model)
+
+    # update "ir_model_fields"."company_dependent" as an identifier for indirect reference
+    cr.execute(
+        """
+        UPDATE ir_model_fields
+           SET company_dependent = TRUE
+         WHERE model = %s
+           AND name = %s
+     RETURNING id
+        """,
+        (model, field)
+    )
+    if not cr.rowcount:
+        # no ir_model_fields, no column
+        remove_column(cr, table, field, cascade=True)
+        return
+    [field_id] = cr.fetchone()
+
+    if default_value is None:
+        where_condition = "{0} IS NOT NULL"
+    else:
+        where_condition = "{0} != " + cr.mogrify("%s", [default_value]).decode()
+    where_condition += format_query(cr, " AND {} IS NOT NULL", company_field)
+
+    using = format_query(cr, "jsonb_build_object({}, {{0}})", company_field)
+    alter_column_type(cr, table, field, "jsonb", using=using, where=where_condition)
+
+    # delete all old default
+    cr.execute("DELETE FROM ir_default WHERE field_id = %s", [field_id,])
```

Comment:
Avoid `ruff` to reformat this on multiple lines.
```suggestion
    cr.execute("DELETE FROM ir_default WHERE field_id = %s", [field_id])
```